File: backend/app/schemas/comparison.py
from pydantic import BaseModel
from typing import List, Optional

class CompareRequest(BaseModel):
    prescription_id_a: int
    prescription_id_b: int

class MedicineSummary(BaseModel):
    id: int
    name: str
    strength: Optional[str] = None
    dosage: Optional[str] = None
    frequency: Optional[str] = None
    duration: Optional[str] = None

class MedicineDiffItem(BaseModel):
    medicine_name: str
    strength_a: Optional[str] = None
    strength_b: Optional[str] = None
    dosage_a: Optional[str] = None
    dosage_b: Optional[str] = None
    frequency_a: Optional[str] = None
    frequency_b: Optional[str] = None
    duration_a: Optional[str] = None
    duration_b: Optional[str] = None

class CompareResponse(BaseModel):
    prescription_a_id: int
    prescription_b_id: int
    added: List[MedicineSummary]
    removed: List[MedicineSummary]
    changed: List[MedicineDiffItem]
    unchanged: List[MedicineSummary]

File: frontend/src/types/comparison.ts
export interface MedicineSummary {
  id: number;
  name: string;
  strength: string | null;
  dosage: string | null;
  frequency: string | null;
  duration: string | null;
}

export interface MedicineDiffItem {
  medicine_name: string;
  strength_a: string | null;
  strength_b: string | null;
  dosage_a: string | null;
  dosage_b: string | null;
  frequency_a: string | null;
  frequency_b: string | null;
  duration_a: string | null;
  duration_b: string | null;
}

export interface CompareResponse {
  prescription_a_id: number;
  prescription_b_id: number;
  added: MedicineSummary[];
  removed: MedicineSummary[];
  changed: MedicineDiffItem[];
  unchanged: MedicineSummary[];
}
